integrations/integrations.go

```diff
@@ -0,0 +1,232 @@
+	"github.com/polymerdao/monomer/genesis"
+	"github.com/polymerdao/monomer/node"
+	"github.com/spf13/viper"
+)
+
+func StartCommandHandler(
+	svrCtx *server.Context,
+	clientCtx client.Context, //nolint:gocritic // hugeParam
+	appCreator servertypes.AppCreator,
+	inProcessConsesus bool,
+	opts server.StartCmdOptions,
+) error {
+	// We assume `inProcessConsensus` is true for now, so let's return an error if it's not.
+	if !inProcessConsesus {
+		return fmt.Errorf("in-process consensus must be enabled")
+	}
+
+	env := environment.New()
+	defer func() {
+		if closeErr := env.Close(); closeErr != nil {
+			svrCtx.Logger.Error(closeErr.Error())
+		}
+	}()
+
+	// --- Do the normal Cosmos SDK stuff ---
+	svrCfg, err := getAndValidateConfig(svrCtx)
+	if err != nil {
+		return fmt.Errorf("failed to get and validate server config: %w", err)
+	}
+
+	app, err := startApp(env, svrCtx, appCreator, opts)
+	if err != nil {
+		return fmt.Errorf("failed to start application: %w", err)
+	}
+
+	// Would usually start a Comet node in-process here, but we replace the
+	// Comet node with a Monomer node.
+	if err := startInProcess(env, svrCtx, svrCfg, clientCtx, app, opts); err != nil {
+		svrCtx.Logger.Error("failed to start Monomer node in-process", "error", err)
+		return fmt.Errorf("failed to start Monomer node in-process: %w", err)
+	}
+
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM)
+	<-sigCh
+
+	return nil
+}
+
+var _ io.WriteCloser = (*fakeTraceWriter)(nil)
+
+type fakeTraceWriter struct{}
+
+func (f *fakeTraceWriter) Write([]byte) (int, error) {
+	return 0, nil
+}
+
+func (f *fakeTraceWriter) Close() error {
+	return nil
+}
+
+// See https://github.com/cosmos/cosmos-sdk/blob/7fb26685cd68a6c1d199dc270c80f49f2bfe7ace/server/start.go#L624
+func startApp(
+	env *environment.Env,
+	svrCtx *server.Context,
+	appCreator servertypes.AppCreator,
+	opts server.StartCmdOptions,
+) (app servertypes.Application, err error) {
+	home := svrCtx.Config.RootDir
+	db, err := opts.DBOpener(home, server.GetAppDBBackend(svrCtx.Viper))
+	if err != nil {
+		return app, err
+	}
+
+	// TODO: Check if is testnet and implement `testnetify` function
+
+	app = appCreator(svrCtx.Logger, db, &fakeTraceWriter{}, svrCtx.Viper)
+	env.Defer(func() {
+		if localErr := app.Close(); localErr != nil {
+			svrCtx.Logger.Error(localErr.Error())
+		}
+	})
+
+	return app, nil
+}
+
+// See https://github.com/cosmos/cosmos-sdk/blob/7fb26685cd68a6c1d199dc270c80f49f2bfe7ace/server/start.go#L307
+// We modify this function to start a Monomer node in-process instead of a Comet node.
+func startInProcess(
+	env *environment.Env,
+	svrCtx *server.Context,
+	_ serverconfig.Config, //nolint:gocritic // hugeParam
+	clientCtx client.Context, //nolint:gocritic // hugeParam
+	app servertypes.Application,
+	opts server.StartCmdOptions,
+) error {
+	g, ctx := getCtx(svrCtx, true)
+
+	svrCtx.Logger.Info("starting Monomer node in-process")
+	// --- Start the Monomer node ---
```

Comment:
nit: we can remove this comment, the function call is pretty self-explanatory (similar for the other `--- ... ---` comments).